casper/src/test/scala/coop/rchain/casper/util/rholang/RuntimeManagerTest.scala

```diff
@@ -27,125 +27,110 @@ import scala.concurrent.duration._
-    val deploy     = ConstructDeploy.sourceDeployNow(badRholang)
-    val (_, Seq(result)) =
-      runtimeManager
-        .use(mgr => computeState(mgr, mgr.emptyStateHash, deploy :: Nil))
-        .runSyncUnsafe(10.seconds)
-
-    result.status.isFailed should be(true)
-    result.cost.cost shouldEqual (accounting.parsingCost(badRholang).value)
+    for {
+      deploy <- ConstructDeploy.sourceDeployNowF(badRholang)
+      result <- runtimeManager.use(computeState(_, deploy))
+      _      = result._2.status.isFailed should be(true)
+      _      = result._2.cost.cost shouldEqual accounting.parsingCost(badRholang).value
+    } yield ()
   }
 
-  it should "charge for parsing and execution" in {
+  it should "charge for parsing and execution" in effectTest {
     val correctRholang = """ for(@x <- @"x"; @y <- @"y"){ @"xy"!(x + y) | @"x"!(1) | @"y"!(2) }"""
-    val deploy         = ConstructDeploy.sourceDeployNow(correctRholang)
-
-    implicit val log: Log[Task]            = Log.log[Task]
-    implicit val metricsEff: Metrics[Task] = new metrics.Metrics.MetricsNOP[Task]
-    implicit val noopSpan: Span[Task]      = NoopSpan[Task]()
-
-    (for {
-      reductionCost <- mkRuntime[Task]("casper-runtime")
-                        .use { runtime =>
-                          implicit val rand: Blake2b512Random = Blake2b512Random(
-                            DeployData.toByteArray(ProtoUtil.stripDeployData(deploy))
-                          )
-                          val initialPhlo = Cost.UNSAFE_MAX
-                          for {
-                            _             <- runtime.reducer.setPhlo(initialPhlo)
-                            term          <- ParBuilderUtil.buildNormalizedTerm[Task](deploy.term)
-                            _             <- runtime.reducer.inj(term)
-                            phlosLeft     <- runtime.reducer.phlo
-                            reductionCost = initialPhlo - phlosLeft
-                          } yield (reductionCost)
-                        }
-      result <- mkRuntimeManager("casper-runtime-manager")
-                 .use {
-                   case runtimeManager =>
-                     for {
-                       state <- computeState(
-                                 runtimeManager,
-                                 runtimeManager.emptyStateHash,
-                                 deploy :: Nil
-                               )
-                       result = state._2.head
-                     } yield (result)
-                 }
-      _           = result.status.isFailed should be(false)
-      parsingCost = accounting.parsingCost(correctRholang)
-    } yield (result.cost.cost shouldEqual ((parsingCost + reductionCost).value)))
-      .runSyncUnsafe(10.seconds)
+
+    runtimeAndManager
+      .use {
+        case (runtime, runtimeManager) => {
+          implicit val rand: Blake2b512Random = Blake2b512Random(32)
```

Comment:
Usually we use `Blake2b512Random(128)`. Is there a reason you have `Blake2b512Random(32)` here?

Reply:
No particular reason. Fixed :)